system/lib/libcxxabi/src/cxa_exception.cpp

```diff
@@ -255,7 +255,11 @@ will call terminate, assuming that there was no handler for the
 exception.
 */
 void
+#ifdef __USING_WASM_EXCEPTIONS__
+__cxa_throw(void *thrown_object, std::type_info *tinfo, void *(*dest)(void *)) {
```

Comment:
oops, this was a TODO for my review 🤣 But I didn't figure out the answer, so the question still stands.

Reply:
Because this pointer is used for destructors and in wasm, destructors return its argument. This comment was added in cxa_exception.hpp, but I'll add that here too.

Reply:
In ARM EHABI destructors also return their argument IIRC. Does ARM use a different signature for `__cxa_throw` too? Or do they just have a different libc++abi API entirely?

Reply:
Hmm, I'm not sure. ARM uses `__cxa_throw` and they don't seem to need this change. I tried deleting the wasm specific part here too and it seems tests are actually working for wasm too. I made this change months ago while debugging some handwritten example which I now don't remember, and I have no idea what changed, but anyway, I'll delete this for now.

Reply:
Oh no, I think we actually need this change. Without this some tests fail with
```
exception thrown: RuntimeError: function signature mismatch
```
I guess ARM was OK because it does not type-check strictly when it calls a function pointer. Reintroduced this change.